### PR TITLE
Wider Surface Head: increase SRF hidden dim 192 → 256

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The SurfaceRefinementHead is the final decoder that maps from the Transolver's hidden representation to surface pressure predictions. It currently uses hidden_dim=192 with 3 layers. Increasing to 256 gives the decoder more capacity to capture the complex nonlinear mapping from latent features to pressure — especially at sharp features (stagnation point, suction peak, pressure recovery).

**Why this should help:** We have 96GB VRAM but only use ~46GB. We're leaving 50GB of VRAM on the table. The SRF head processes only surface nodes (~2-5K per sample, much smaller than the full mesh ~10-20K), so increasing its width has modest compute overhead. More decoder capacity is one of the simplest scaling levers available.

**Evidence:** Wider decoders are a reliable improvement in transformer architectures when the bottleneck is in the output projection. The KAN experiment (#2325) showed that changing the activation type doesn't help — but simply having more parameters in the same architecture may.

**Compute impact:** Increasing from 192→256 adds ~40% more parameters to the SRF head only. Per-step overhead should be <5% since the SRF head is a small fraction of total compute.

## Instructions

This is a purely hyperparameter change — no code modifications needed.

### Run 2 seeds with surface_refine_hidden=256

```bash
cd cfd_tandemfoil && python train.py \
  --agent edward --wandb_name "edward/srf256-s42" --seed 42 \
  --wandb_group wider-surface-head \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 150 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 256 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature --re_stratified_sampling

# Repeat for seed 73: --seed 73 --wandb_name "edward/srf256-s73"
```

The ONLY change from baseline is `--surface_refine_hidden 256` (was 192).

Run both seeds in parallel:
```bash
CUDA_VISIBLE_DEVICES=0 python train.py ... --seed 42 &
CUDA_VISIBLE_DEVICES=1 python train.py ... --seed 73 &
wait
```

## Baseline

| Metric | Baseline | Target to beat |
|--------|----------|----------------|
| p_in | 11.742 | < 11.742 |
| p_oodc | 7.643 | < 7.643 |
| p_tan | 27.874 | < 27.874 |
| p_re | 6.419 | < 6.419 |

- **Baseline W&B runs:** k5qwvce4 (seed 42), 7oa5xfhi (seed 73)
- **val/loss baseline:** ~0.37
- **Baseline SRF hidden:** 192

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2333 because Wider SRF head (192→256) regresses all 4 metrics: p_in +2.3%, p_oodc +2.3%, p_tan +3.9%, p_re +0.6%. The additional parameters overfit rather than improving capacity. The current SRF hidden size of 192 is well-tuned. W&B: 3ex38gxd (s42), 22uz9576 (s73)..

**Comment by tcapelle** [comment]
STUDENT edward:

## Results

### Seed 42 — W&B run `3ex38gxd`
- **Best epoch:** 147 (wall-clock timeout at 180 min)
- **val/loss:** 0.3823
- **Surface MAE (p):**
  - p_in: 12.1 (baseline 11.742, **+2.6% worse**)
  - p_tan: 28.8 (baseline 27.874, **+3.3% worse**)
  - p_oodc: 7.9 (baseline 7.643, **+3.4% worse**)
  - p_re: 6.5 (baseline 6.419, **+1.3% worse**)
- **Peak VRAM:** 46.3 GB
- **Command:**
  ```
  CUDA_VISIBLE_DEVICES=0 python train.py --agent edward --wandb_name "edward/srf256-s42" --seed 42 \
    --wandb_group wider-surface-head \
    --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
    --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
    --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
    --cosine_T_max 150 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
    --pressure_first --pressure_deep \
    --residual_prediction --surface_refine --surface_refine_hidden 256 --surface_refine_layers 3 \
    --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
    --gap_stagger_spatial_bias \
    --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
    --te_coord_frame --wake_deficit_feature --re_stratified_sampling
  ```

### Seed 73 — W&B run `22uz9576`
- **Best epoch:** 146 (wall-clock timeout at 181 min)
- **val/loss:** 0.3815
- **Surface MAE (p):**
  - p_in: 12.0 (baseline 11.742, **+2.0% worse**)
  - p_tan: 29.1 (baseline 27.874, **+4.4% worse**)
  - p_oodc: 7.7 (baseline 7.643, **+0.7% worse**)
  - p_re: 6.4 (baseline 6.419, **-0.3% — neutral**)
- **Peak VRAM:** 46.0 GB
- **Command:** Same as above with `--seed 73 --wandb_name "edward/srf256-s73"` on `CUDA_VISIBLE_DEVICES=1`

### Summary

| Metric | Baseline | Seed 42 | Seed 73 | Δ (avg) |
|--------|----------|---------|---------|---------|
| p_in   | 11.742   | 12.1    | 12.0    | +2.3% worse |
| p_tan  | 27.874   | 28.8    | 29.1    | +3.9% worse |
| p_oodc | 7.643    | 7.9     | 7.7     | +2.0% worse |
| p_re   | 6.419    | 6.5     | 6.4     | +0.5% worse |
| val/loss | ~0.37  | 0.3823  | 0.3815  | ~+3% worse |

### What happened

**The wider SRF head (256 vs baseline 192) degraded performance across all surface pressure metrics.** Both seeds are consistent: every metric is worse, with p_tan (+3.9%) and p_in (+2.3%) showing the largest regressions. val/loss also increased slightly (~0.38 vs ~0.37 baseline).

VRAM increased modestly (46 GB vs ~45 GB baseline), confirming the parameter count increase was small as expected. The issue is not compute — the extra capacity simply didn't help.

**Why it didn't work:**
- The SRF head at 192 dims was likely already sufficient for the mapping complexity. Adding more parameters to the decoder without more informative input features just adds parameters to overfit on noise.
- The Transolver backbone (hidden_dim=192) feeds the SRF head. Widening the head to 256 while keeping the backbone at 192 creates an information bottleneck mismatch — you can't extract more signal from a 192-dim bottleneck by widening the downstream MLP.
- This is consistent with the KAN experiment (#2325) finding: the decoder activation type didn't matter either. The bottleneck is upstream.

### Suggested follow-ups
- **Widen the backbone** (hidden_dim > 192) rather than the decoder — that's where the actual capacity bottleneck is, though the compute cost is much higher.
- **Deeper SRF head** (e.g. 5 layers at 192 dim) might help more than wider, as depth adds representational complexity without the bottleneck mismatch.
- Possibly explore skip connections or residual blocks in the SRF head to help gradient flow with deeper architectures.
